operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,342 @@
+    /* (non-Javadoc)
+     * @see com.ibm.fhir.server.operation.spi.AbstractOperation#buildOperationDefinition()
+     */
+    @Override
+    protected OperationDefinition buildOperationDefinition() {
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream(OPERATION_DEFINITION_FILE)) {
+            return FHIRParser.parser(Format.JSON).parse(in);
+        } catch (FHIRParserException e) {
+            throw new Error("There has been a parsing error in the operation definition of the $everything operation.", e);
+        } catch (IOException e) {
+            throw new Error("There has been an IO error reading the operation definition of the $everything operation.", e);
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see com.ibm.fhir.server.operation.spi.AbstractOperation#doInvoke(com.ibm.fhir.server.operation.spi.FHIROperationContext, java.lang.Class, java.lang.String, java.lang.String, com.ibm.fhir.model.resource.Parameters, com.ibm.fhir.server.operation.spi.FHIRResourceHelpers)
+     */
+    @Override
+    protected Parameters doInvoke(FHIROperationContext operationContext, Class<? extends Resource> resourceType, String logicalId, String versionId, Parameters parameters, FHIRResourceHelpers resourceHelper) throws FHIROperationException {
+        LOG.entering(this.getClass().getName(), "doInvoke");
+
+        Patient patient = null;
+        try {
+            patient = (Patient) resourceHelper.doRead(PATIENT, logicalId, false, false, null, null);
+        } catch (Exception e) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("An unexpected error occurred while reading patient " + logicalId, IssueType.EXCEPTION);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+        if (patient == null) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("Patient with ID " + logicalId + " does not exist.", IssueType.NOT_FOUND);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+
+        Entry patientEntry = buildPatientEntry(operationContext, patient);
+        List<Entry> allEntries = new ArrayList<>(DEFAULT_RESOURCE_COUNT * 2);
+        allEntries.add(patientEntry);
+        
+        // We can't always use the "date" query parameter to query by clinical date, only with some resources.
+        // Initial list obtained from the github issue: https://github.com/IBM/FHIR/issues/1044#issuecomment-769788097
+        // Otherwise the search throws an exception. We create a params map with and without and use as needed
+        MultivaluedMap<String, String> queryParameters = parseQueryParameters(parameters);
+        MultivaluedMap<String, String> queryParametersWithoutDates = new MultivaluedHashMap<String, String>(queryParameters);
+        queryParametersWithoutDates.remove(DATE_QUERY_PARAMETER);
```

Comment:
Parameters parameters and is mediate as a post or get. 
I suggest using the Parameters object as you don't have to jointly care about the input method.

Reply:
@prb112 not getting this can you elaborate?

Reply:
All of your query parameters are validated at this point and pushed into Parameters parameters object. 
You should probably access this object to get the queryParameters (or more directly the Parameters value) 

Look at FHIROperationUtil to see how it is generated.

Reply:
I am using the validated `Parameter` from the `Parameters` object at this point to extract the parameters of the $everything operation and mapping those to the corresponding `MultivalueMap` query parameters that the `doSearch()` method will understand. Let me know if anything other than that is necessary. Thanks!